pkg/control/sidecarcontrol/util.go

```diff
@@ -103,6 +103,10 @@ func PodMatchedSidecarSet(c client.Client, pod *corev1.Pod, sidecarSet *appsv1al
 	if err != nil {
 		return false, err
 	}
+	// if no selector set, return true
+	if selector.String() == "" {
```

Comment:
if selector.Empty() {

Reply:
selector.Empty() return false when no selector set, so I turned to selector.String() == ""